migration-guide.md

```diff
@@ -0,0 +1,64 @@
+# Swift Migration Guide
+
+This guide should help you with migrating projects on previous version of Swift to newer versions.
+
+### Swift 2 → Swift 3
+
+**IMPORTANT:** Make granular commits after each step and substep in this guide. Make sure the commit messages follow the proper rules.
+
+1. Clone project and pull/checkout all branches ([script here](http://stackoverflow.com/a/21189710/1001803)).
+    - Clean up all branches and keep only `master` and `develop`
+    - Make sure `develop` is even or ahead of `master`
+2. From `develop` branch run `git checkout -b feature/swift3-migration`.
+3. *(optional)* Clean project, remove unused files, update file structure.
+4. Update `Cartfile` to point to Swift 3 versions of dependencies. 📦 
+    - Update framework names if used in project
+        - `NStack` → `NStackSDK`
+        - `Serializable` → `Serpent`
+    - Remove existing Carthage folder and run `carthage update --platform ios` (change platform as needed)
+5. *(optional)* Update translations generator.
+    - Download from latest Swift 3 release [on Github](https://github.com/nodes-ios/nstack-translations-generator/releases), do NOT add to Xcode
+    - Remove any old translations generator and templates present in project
+6. Open `YourProject.xcodeproj` in Xcode 8 and run Swift migrator.
+    - Go through suggested changes to double check
+7. Go to Build Phases and modify current run scripts.
+    - Update translations script as per [README](https://github.com/nodes-ios/nstack-translations-generator)
+    - Add or update script checking for TODO & FIXME ([gist here](https://gist.github.com/nickskull/2df890a021ada999bda383b8413c9473))
+    - Add or update Carthage copy frameworks script with new names/frameworks
+8. Update build settings.
+    - Enable whole module optimizations for release/adhoc
+    - Enable bitcode if possible
+    - Remove unused/old user-defined build settings (scroll to the bottom of build settings)
+9. Build and fix errors, rinse & repeat until project compiles. 😬
+    - **Alamofire** 
+	    - Update completion type to `DataResponse`
+	    - Rename completion closure paramter to response instead of result
+	    - Switch on `response.result` instead of result
+	    - Update switch cases to `.success(let data)` and `.failure(let error)`
```

Comment:
Also need to manually rearrange the request calls, as the URL comes first now and then the HTTP method is after.